sdkcompat/v223/com/google/idea/sdkcompat/general/BaseSdkCompat.java

```diff
@@ -0,0 +1,179 @@
+    // API to create an IdeModifiableModelsProvider.
+    return new IdeModifiableModelsProviderImpl(project);
+  }
+
+  /** #api212: inline into BlazeNewProjectWizard */
+  public static void setContextWizard(WizardContext context, AbstractWizard<?> wizard) {
+    context.putUserData(AbstractWizard.KEY, wizard);
+  }
+
+  /** #api211: inline into HgConfigurationProjectPanel. Method params changed in 2021.2.4 */
+  public static void reset(
+      VcsExecutablePathSelector executablePathSelector,
+      @Nullable String globalPath,
+      boolean pathOverriddenForProject,
+      @Nullable String projectPath,
+      String autoDetectedPath) {
+    executablePathSelector.reset(globalPath, pathOverriddenForProject, projectPath);
+    executablePathSelector.setAutoDetectedPath(autoDetectedPath);
+  }
+
+  /** #api212: inline this method. */
+  @SuppressWarnings("rawtypes")
+  public static boolean isIncrementalRefreshProperty(VcsLogProperty property) {
+    return property == VcsLogProperties.SUPPORTS_INCREMENTAL_REFRESH;
+  }
+
+  /** #api213: inline this method */
+  @Nullable
+  public static String getIdeRestarterPath() {
+    Path startFilePath = Restarter.getIdeStarter();
+    return startFilePath == null ? null : startFilePath.toString();
+  }
+
+  /** #api213: inline into IndexingLogger */
+  public static JsonDuration getTotalIndexingTime(
+      JsonFileProviderIndexStatistics providerStatisticInput) {
+    return providerStatisticInput.getTotalIndexingVisibleTime();
+  }
+
+  /** #api213: inline this method. */
+  public static String getLibraryNameFromLibraryOrigin(LibraryOrigin libraryOrigin) {
+    // TODO(b/230430213): adapt getLibraryNameFromLibraryOrigin to work in 221
+    return "";
+  }
+
+  /** #api213: Inline into KytheRenameProcessor. */
+  public static RenamePsiElementProcessor[] renamePsiElementProcessorsList() {
+    ArrayList<RenamePsiElementProcessor> result = new ArrayList<>();
+    for (RenamePsiElementProcessorBase processor :
+        RenamePsiElementProcessor.EP_NAME.getExtensions()) {
+      if (processor instanceof RenamePsiElementProcessor) {
+        result.add((RenamePsiElementProcessor) processor);
+      }
+    }
+    return result.toArray(new RenamePsiElementProcessor[0]);
+  }
+
+  /** #api213: Inline into WorkspaceFileTextField . */
+  public static LocalFsFinder.VfsFile getVfsFile(VirtualFile file) {
+    return new LocalFsFinder.VfsFile(file);
+  }
+
+  /** #api213: Inline into WorkspaceFileTextField . */
+  public static FileLookup.LookupFile getIoFile(Path path) {
+    return new LocalFsFinder.IoFile(path);
+  }
+
+  /** #api213: Inline into BlazeProjectCreator. */
+  public static OpenProjectTask createOpenProjectTask(Project project) {
+    return OpenProjectTask.build().withProject(project);
+  }
+
+  public static Project openProject(VirtualFile projectSubdirectory, Project projectToClose, boolean forceOpenInNewFrame) {
```

Comment:
Please add this comment: `/** #api213 inline when this api support is dropped */`